spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/MetricsReplicationEnvironmentPostProcessor.java

```diff
@@ -280,13 +262,16 @@ private void traversePropertyClassFields(Class<?> metricsPropertyClass, Consumer
 	private void traverseClassFieldsRecursively(Class<?> metricsPropertyClass, String metricsPrefix,
 			Consumer<String> metricsReplicationHandler) {
 		for (Field field : metricsPropertyClass.getDeclaredFields()) {
-			if (field.getType().isMemberClass() && Modifier.isStatic(field.getType().getModifiers())) {
+			var isStaticMemberClass = field.getType().isMemberClass() && Modifier.isStatic(field.getType().getModifiers());
+			if (isStaticMemberClass && !field.getType().isEnum()) {
```

Comment:
I found an unrelated bug that has been in this class since its inception... enum fields that are also nested static members (e.g. `WavefrontProperties.apiTokenType` where the `ApiTokenType` is defined in the config props class). It would just die in the recursion and was not visible to end user other than missing expected replicated properties.

Reply:
Good catch